apps/server/src/modules/system/service/system.service.ts

```diff
@@ -15,13 +16,17 @@ export class SystemService {
 		} else {
 			systemEntities = await this.systemRepo.findByFilter(type, onlyOauth);
 		}
-		const systemDtos: SystemDto[] = SystemMapper.mapFromEntitiesToDtos(systemEntities);
-		return systemDtos;
+		const keycloakConfig = (await this.systemRepo.findByFilter(SysType.KEYCLOAK, true))[0];
+		systemEntities.forEach((systemEntity) => {
+			if (keycloakConfig && systemEntity.type === SysType.OIDC && !systemEntity.oauthConfig) {
+				systemEntity.oauthConfig = keycloakConfig.oauthConfig;
+			}
+		});
+		return SystemMapper.mapFromEntitiesToDtos(systemEntities);
 	}
 
 	async findById(id: EntityId): Promise<SystemDto> {
-		const entity: System = await this.systemRepo.findById(id);
-		const systemDto: SystemDto = SystemMapper.mapFromEntityToDto(entity);
-		return systemDto;
+		const entity = await this.systemRepo.findById(id);
```

Comment:
If this system has type OIDC and no oauthConfig, does need to use the config of keycloak as in `find`?